pinot-integration-tests/src/test/java/org/apache/pinot/integration/tests/DefaultCommitterRealtimeIntegrationTest.java

```diff
@@ -0,0 +1,204 @@
+
+    setUpRealtimeTable(avroFiles.get(0));
+    buildSegment();
+  }
+
+  @Override
+  protected boolean useLlc() {
+    return true;
+  }
+
+  @Test
+  public void testDefaultCommitter() throws Exception {
+    IndexLoadingConfig indexLoadingConfig = new IndexLoadingConfig();
+    ServerMetrics serverMetrics = new ServerMetrics(new MetricsRegistry());
+    ServerSegmentCompletionProtocolHandler
+        protocolHandler = new ServerSegmentCompletionProtocolHandler(serverMetrics, getTableName());
+
+    SegmentCompletionProtocol.Response prevResponse = new SegmentCompletionProtocol.Response();
+    LLRealtimeSegmentDataManager.SegmentBuildDescriptor segmentBuildDescriptor = mock(
+        LLRealtimeSegmentDataManager.SegmentBuildDescriptor.class);
+
+    RealtimeSegmentZKMetadata metadata = _helixResourceManager.getRealtimeSegmentMetadata(getTableName()).get(0);
+
+    String instanceId = _helixResourceManager.getAllInstances().get(0);
+
+    SegmentCompletionProtocol.Request.Params params = new SegmentCompletionProtocol.Request.Params();
+    params.withSegmentName(metadata.getSegmentName()).withInstanceId(instanceId).withOffset(END_OFFSET);
+
+    _realtimeSegmentUntarred = new File(_indexDir.getParentFile(), metadata.getSegmentName());
+    FileUtils.copyDirectory(_indexDir, _realtimeSegmentUntarred);
+
+    TarGzCompressionUtils.createTarGzOfDirectory(_realtimeSegmentUntarred.getAbsolutePath());
+
+    when(segmentBuildDescriptor.getSegmentTarFilePath()).thenReturn(_realtimeSegmentUntarred + TARGZ_SUFFIX);
+    when(segmentBuildDescriptor.getBuildTimeMillis()).thenReturn(0L);
+    Map<String, File> metadataPaths = new HashMap();
+    metadataPaths.put(metadata.getSegmentName(), new File(_indexDir, METADATA_FILE_NAME));
+
+    when(segmentBuildDescriptor.getMetadataFiles()).thenReturn(metadataPaths);
+    when(segmentBuildDescriptor.getOffset()).thenReturn(END_OFFSET);
+    when(segmentBuildDescriptor.getSegmentSizeBytes()).thenReturn(0L);
+    when(segmentBuildDescriptor.getWaitTimeMillis()).thenReturn(0L);
+
+    // Get realtime segment name
+    String segmentList = sendGetRequest(_controllerRequestURLBuilder.forSegmentListAPIWithTableType(getTableName(), "REALTIME"));
+    JsonNode realtimeSegmentsList =
+        getSegmentsFromJsonSegmentAPI(segmentList, CommonConstants.Helix.TableType.REALTIME.toString());
+    String segmentName = realtimeSegmentsList.get(0).asText();
+
+    // Send segmentConsumed request
+    sendGetRequest("http://localhost:" + DEFAULT_CONTROLLER_PORT +
+            "/segmentConsumed?instance=" + instanceId + "&name=" + segmentName + "&offset=" + END_OFFSET);
+
+    SegmentCommitterFactory
+        segmentCommitterFactory = new SegmentCommitterFactory(LOGGER, indexLoadingConfig, protocolHandler);
+    SegmentCommitter segmentCommitter = segmentCommitterFactory.createDefaultSegmentCommitter(params);
+    segmentCommitter.commit(END_OFFSET, 3, segmentBuildDescriptor);
+  }
+
+  public void buildSegment()
```

Comment:
can we create a segment and check-in? How long does it take for the test to run?

Reply:
8 seconds, my segment is built with only one row, so I think it should be pretty fast - thoughts?

Reply:
8 seconds seems reasonable. Let us leave it like that then.